src/server/pfs/server/driver.go

```diff
@@ -1461,6 +1461,15 @@ func (d *driver) squashCommit(txnCtx *txncontext.TransactionContext, userCommit
 			if commit == nil {
 				return errors.Errorf("encountered nil parent commit in %s@%s...%s", lowerKey, lower.ID, upper.ID)
 			}
+			// are we leaking commits if we squash a commit with subvenance that forks and then merges?
```

Comment:
@brycemcanally I may be misunderstanding the code, but this comment I made looks to me like it could be an issue.

In the diagram below it looks to me that within a subvenance commit range we would trace back from one upper commit to the lower, but would have deleted other commits along the path that needs to be taken for another deletion.

Does that make sense? I would expect the algorithm to queue and process multiple parent commits

Reply:
@brycemcanally after our conversation earlier, this looks to me like a non-issue. Since the commit path is a tree, if this delete closure is called on all HEAD leaves, then there shouldn't be any undeleted commits from most descendant to to ascendant commits.

Reply:
Yeah, we can revisit this after global IDs are in because this will probably change.